keps/sig-node/2008-forensic-container-checkpointing/README.md

```diff
@@ -292,14 +419,111 @@ Checkpointing containers will be possible again.
 
 ###### Are there any tests for feature enablement/disablement?
 
-Currently no.
+Currently the test will automatically be skipped if the feature is not enabled.
+Tests will be extended to explicitly test if the feature is disabled as well
+as if it is enabled.
+
+### Rollout, Upgrade and Rollback Planning
+
+If it is not enabled via the feature gate it will return `404 page not found`.
+If it is not enabled in the underlying container engine a `500` will be returned
+with an error message from the container engine. If it is enabled the API
+endpoint exists if disabled then it does not exist. No planning necessary.
+
+Documented at <https://kubernetes.io/docs/reference/node/kubelet-checkpoint-api/>
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+The feature depends on the existence of the `CheckpointContainer` CRI RPC.
+If the underlying container engine does not support or if it is not implemented,
+the kubelet API endpoint will fail with `500`. The error code is the same if the
+container engine does not implement it explicitly or if the underlying container
+engine is too old. The difference between does two failures are only visible in
+the error message returned from the container engine.
+
+If the underlying container engine does not support the CRI RPC the kubelet API endpoint
+will always return `500`.
+
+It cannot directly impact running workloads, but if the *kubelet* API endpoint is
+called if the underlying container engine does no longer support it, the checkpoint
+request will fail.
+
+###### What specific metrics should inform a rollback?
+
+CRI metrics will be added to track checkpointing failures to inform a rollback
+decision.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+No data is stored, so re-enabling starts from a clean slate.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No.
+
+### Monitoring Requirements
+
+Querying the state of the feature gate offers the possibility to detect
+if the API endpoint will return `404` or not.
+
+<!--
+This section must be completed when targeting beta to a release.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+
+###### How can an operator determine if the feature is in use by workloads?
+
+As it is not exposed in the Kubernetes API it cannot be determined. This is
+only visible in the kubelet. Also, this is a feature workloads are not using
+directly, but which is only external entities can trigger. Access to the
+*kubelet* API endpoints is needed. It might be detectable if operators can
+query the state of different feature gates. An operator could also use metrics
```

Comment:
tell us which ones.  I think you found  a reference to them.

Reply:
added